### PR TITLE
Modal sidebar

### PR DESCRIPTION
**Pull request by Aringoaway**
refactor sidebar for mobile version

### CODE REVIEW

**Comment by Aringoaway** [comment]
@Pe5h4 can you review pls

sidebar

File: src/containers/Header/index.js [hunk 1]
```diff
@@ -16,11 +16,46 @@ export function Header(props) {
...
 
 	const [headerProperties, setHeaderProperties] = useState(false);
+	const { width } = useWindowDimensions();
+
+	function useWindowDimensions() {
+		const [windowDimensions, setWindowDimensions] = useState(
+			getWindowDimensions()
+		);
+
+		useEffect(() => {
```
Comment (Pe5h4) [inline src/containers/Header/index.js:26]: @Aringoaway You should not use hooks in nested functions: https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

Instead you can use `useCallback` component https://reactjs.org/docs/hooks-reference.html#usecallback

File: src/containers/Sidebar/index.js [hunk 2]
```diff
@@ -47,28 +49,82 @@ const Sidebar = (props) => {
 		return itemsList;
 	}, [navConfig])
 
+	function useWindowDimensions() {
+		const [windowDimensions, setWindowDimensions] = useState(
+			getWindowDimensions()
+		);
+
+		useEffect(() => {
```
Comment (Pe5h4) [inline src/containers/Sidebar/index.js:57]: @Aringoaway The same as I wrote above about the rule of not calling useEffect in nested function. It should be always top level

File: src/styles/components/sidebar.scss [hunk 3]
```diff
@@ -195,3 +205,97 @@ $sidebar-item-font-size: 14px;
...
+		visibility: visible;
+		font-size: 1.5rem;
+		margin-left: auto !important;
+		cursor: pointer;
+		position: relative;
+
+		& [class^="cil-"] {
+			position: absolute;
+			top: 20px;
+			left: 32px;
```
Comment (Pe5h4) [inline src/styles/components/sidebar.scss:267]: @Aringoaway To me it looks like the hamburger menu has bigger left padding than the `chevron` icon's (opening the collapsed header items) right padding. There is a disproportion.

**Review by Pe5h4 — CHANGES_REQUESTED** [review]

File: src/styles/components/sidebar.scss (hunk 3 above)
Comment (Aringoaway) [reply src/styles/components/sidebar.scss]: @Pe5h4 fixed it

**Comment by Aringoaway** [comment]
@Pe5h4 can you review again please, I fixed your comments
sidebar-fixed

File: src/containers/Sidebar/index.js [hunk 4]
```diff
@@ -47,28 +50,69 @@ const Sidebar = (props) => {
 		return itemsList;
 	}, [navConfig])
 
+	useEffect(() => {
```
Comment (Pe5h4) [inline src/containers/Sidebar/index.js:53]: @Aringoaway useEffect should be always on the top of the file, so please move it to line 19

File: src/styles/components/sidebar.scss [hunk 5]
```diff
@@ -1,3 +1,13 @@
+:root[data-theme="light"] {
+	--toggle-text-color: #828eb0;
+
+}
+
+:root[data-theme="dark"] {
+	--toggle-text-color: #39487a;
+}
+
+$toggle-text-color: var(--toggle-text-color);
```
Comment (Pe5h4) [inline src/styles/components/sidebar.scss:10]: @Aringoaway Cannot this be loaded directly from asab-ui styles? I mean the toggle text color

**Comment by Pe5h4** [comment]
@Aringoaway Small issues, otherwise it looks good!

File: src/containers/Header/index.js [hunk 6]
```diff
@@ -16,11 +16,33 @@ export function Header(props) {
...
 		<header className={`application-header ${headerProperties ? 'header-props-open' : ""}`}>
 			{(props.app.props.hasSidebar || typeof props.app.props.hasSidebar == 'undefined') ?
 				<>
+					{windowDimensions.width < 800 &&
+						<div className="mobile-logo-position">
+							<Link to={href}>
+								<img
+									src={props.brand_image.full}
+									alt={props.title}
+									width="150"
```
[the remark this answers is not in the document]
Comment (Aringoaway) [reply src/containers/Header/index.js:40]: @Pe5h4 Yes I'm sure. I just copied it from Sidebar

**Review by Pe5h4 — APPROVED** [review]
So far it looks fine, but I would like to see how does it look like in dark mode. Also in the next iteration, **we should implement breadcrumbs** for small screens. And I am still not convinced about the logo size (in overall 150x50 and 50x50 looks too large) - if we will decide to leave it 150x50 and 50x50, it has to be changed in documentation as well.

**Comment by Aringoaway** [comment]
@Pe5h4 dark mode and small logo 120x30

dark-mode-sidebar

**Comment by Aringoaway** [comment]
@Pe5h4 

how it looks:

modal-sidebar

File: src/styles/components/header.scss [hunk 7]
```diff
@@ -20,6 +18,13 @@ $nav-bar-link: var(--nav-bar-link);
 	padding: 0.5rem 1rem;
 	background-color: $header-bg;
 
+	.mobile-logo-position {
```
Comment (Pe5h4) [inline src/styles/components/header.scss:21]: @Aringoaway The logo looks to me a bit shifted to the right. Is there any option how to center it without using too much styling as you have below? What I mean something like `text-align: -webkit-center` maybe? But I am just guessing about that suggested option...

File: src/styles/components/sidebar.scss [hunk 8]
```diff
@@ -195,3 +196,97 @@ $sidebar-item-font-size: 14px;
...
+			left: 0;
+		}
+	}
+}
+
+@media screen and (max-width: 800px) {
+	.mobile-sidebar-burger {
+		color: $toggle-text-color;
+		visibility: visible;
+		font-size: 1.5rem;
```
Comment (Pe5h4) [inline src/styles/components/sidebar.scss:250]: @Aringoaway Where did you take this icon size? As far as I know in Header, we have icons `1.1rem`

**Review by Pe5h4 — CHANGES_REQUESTED** [review]

File: src/styles/components/sidebar.scss (hunk 8 above)
Comment (Pe5h4) [reply src/styles/components/sidebar.scss:250]: I think it should be the same everywhere (thus also in this toggler arrow for header dropdown)

Comment (Pe5h4) [reply src/styles/components/sidebar.scss:250]: If possible, please change it there as well

Comment (Aringoaway) [reply src/styles/components/sidebar.scss:250]: @Pe5h4  It's a size of toggle button, 1.5 rem
image

Comment (Pe5h4) [reply src/styles/components/sidebar.scss:250]: @Aringoaway I know, that's why I wrote you to change it as well. We need to have some consistency in item sizes, right? But lets discuss that on standup

**Comment by Aringoaway** [comment]
@Pe5h4 I fix logo without sidebar. Can you review pls?
sidevar

**Review by Pe5h4 — APPROVED** [review]
Ok, it looks gut!

File: CHANGELOG.md [hunk 9]
```diff
@@ -283,6 +283,8 @@
 
 - Implement search, filter and sort by to Microservices component (INDIGO Sprint 220712, [!330](https://github.com/TeskaLabs/asab-webui/pull/330)
 
+- Refactor Sidebar for mobile screen (INDIGO Sprint 220712, [!327](https://github.com/TeskaLabs/asab-webui/pull/327)
```
Comment (Pe5h4) [inline CHANGELOG.md:286]: @Aringoaway btw. this sprint is `220722`

**Pull request merged**